tests/publiccloud/ssh_interactive_init.pm

```diff
@@ -39,49 +34,44 @@ sub run {
+    unless (get_var('PUBLIC_CLOUD_QAM') && get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
+        $instance->run_ssh_command(cmd => qq(echo -e "$testapi::password\\n$testapi::password" | sudo passwd root));
+    }
+
+    # Permit root passwordless login over SSH
     $instance->run_ssh_command(cmd => 'sudo sed -i "s/PermitRootLogin no/PermitRootLogin prohibit-password/g" /etc/ssh/sshd_config');
-    $instance->run_ssh_command(cmd => "sudo mkdir -p /root/.ssh");
-    $instance->run_ssh_command(cmd => "sudo chmod -R 700 /root/.ssh");
-    $instance->run_ssh_command(cmd => 'sudo cp /home/' . $instance->username() . '/.ssh/authorized_keys /root/.ssh/');
-    $instance->run_ssh_command(cmd => 'sudo chmod 644 /root/.ssh/authorized_keys');
-    $instance->run_ssh_command(cmd => 'sudo chown root /root/.ssh/authorized_keys');
     $instance->run_ssh_command(cmd => 'sudo systemctl reload sshd');
 
-    # Create normal user on remote instance and make it accessible by key
-    $instance->run_ssh_command(cmd => "sudo useradd " . $testapi::username);
-    $instance->run_ssh_command(cmd => "sudo mkdir /home/" . $testapi::username);
-    $instance->run_ssh_command(cmd => "sudo chown -R " . $testapi::username . " /home/" . $testapi::username . "/");
-    $instance->run_ssh_command(cmd => 'echo -e "' . $testapi::password . '\n' . $testapi::password . '" | sudo passwd ' . $testapi::username);
-    $instance->run_ssh_command(cmd => "sudo mkdir /home/" . $testapi::username . "/.ssh");
-    $instance->run_ssh_command(cmd => "sudo chmod -R 700 /home/" . $testapi::username . "/.ssh");
-    $instance->run_ssh_command(cmd => 'sudo cp /home/' . $instance->username() . '/.ssh/authorized_keys /home/' . $testapi::username . '/.ssh/');
-    $instance->run_ssh_command(cmd => "sudo chmod 644 /home/" . $testapi::username . "/.ssh/authorized_keys");
-    $instance->run_ssh_command(cmd => "sudo chown -R " . $testapi::username . " /home/" . $testapi::username . "/.ssh/");
-
-    # Print ssh_serial_ready to serial console when interactive ssh session is initiated
-    $instance->run_ssh_command(cmd => "echo \"if tty -s; then echo ssh_serial_ready >> /dev/sshserial; fi\" > .bashrc");
-    $instance->run_ssh_command(cmd => "sudo chmod 777 .bashrc");
-    $instance->run_ssh_command(cmd => "sudo cp .bashrc /root/");
-    $instance->run_ssh_command(cmd => "sudo cp .bashrc /home/" . $testapi::username . "/");
+    # Copy SSH settings for remote root
+    $instance->run_ssh_command("sudo install -o root -g users -m 0700 -dD /root/.ssh");
+    $instance->run_ssh_command("sudo install -o root -g users -m 0644 /home/$instance->{username}/.ssh/authorized_keys /root/.ssh/");
+
+    # Create remote user and set him a password
+    $instance->run_ssh_command(cmd => "sudo useradd -m $testapi::username");
```

Comment:
now I lost . how what we doing here correlate with what we doing [here](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12405/files#diff-50203eb60c47725b927bbc067479f415fea44c62d5e08b9f23ee7015042387d2R46)

Reply:
There we prepare the user of the OSD helper VM so if we switch to `user-console` we can also `ssh -t sut` there.
And here we prepare the remote user so we have both `root` and `bernhard` on the SUT.

Reply:
ah I didn't realize that "there" commands executed not in `run_ssh_command` . thanks , make sense

Reply:
I tried to explain it in comments but if you find better phrasing please reopen